### DISCUSSION TITLE
Scikit-learn SGD vs RiverML

### DISCUSSION THREAD

**Discussion by numersoz** [online-ml/river#1194] [Q&A] [answered]
Can one replicate RiverML SGD online learning using scikit-learn SGD `https://scikit-learn.org/stable/modules/sgd.html` ?

If so, how would one approach to formulating, to replicate similar online learning concept of RiverlML?

**Comment by MaxHalford** [comment, marked as answer]
Hey there @numersoz. Yes, it's definitely possible to get the exact same results with scikit-learn and River. In fact, there are several unit tests which ensure this. Check them out [here](https://github.com/online-ml/river/blob/main/river/linear_model/test_glm.py#L208). We test linear regression, logistic regression, perceptron, L1 regu, mini-batching, etc.

One downside of scikit-learn's implementation is that the number of features is static. You can't add or remove features between mini-batches. But maybe that's a problem for you.
